nir/src/main/scala/scala/scalanative/nir/Linktime.scala

```diff
@@ -0,0 +1,49 @@
+package scala.scalanative.nir
+
+sealed trait LinktimeCondition {
+  def position: Position
+}
+
+object LinktimeCondition {
+
+  case class SimpleCondition(name: Global, comparison: Comp, value: Val)(
```

Comment:
Instead of storing a reference to the definition of the condition here, consider directly storing the string literal that is in its annotation. For example if calling
```scala
@linkTimeResolved("foo")
def bar: String = ...

if (bar == "hello")
```
instead of storing
1. a member of `bar` that references `"foo"`, and
2. a `SimpleCondition(Global(bar), ...)`

consider storing
1. nothing for the *definition* of `bar` (not present in the NIR at all)
2. a `SimpleCondition("foo", ...)`

This would further simplify several things, IMO, and notably the fact that we do not need to enhance arbitrary `Global`s in the NIR with the possibility of being a link-time-resolved value.

Reply:
Good idea, it's a much better approach, since NIR global struct consistent of only a single field. 
Fixed that and PR is ready for next review round